### PR TITLE
Removed "peak device memory" metric

### LINKED ISSUE #8337: [FEA] Remove "peak device memory" metric
**Is your feature request related to a problem? Please describe.**
Currently, the "peak device memory" metric is used to monitor the peak device memory when reading files in spark-rapids, but there isn't a specific use case for it. So we propose removing this metric from the codebase completely to simplify our codebase and avoid any confusion in the future.

**Describe the solution you'd like**
Remove it completely from codebase.

**Describe alternatives you've considered**
Hide it from user interface only.

**Additional context**
Related issue: #8297

### PR DESCRIPTION
**Pull request by thirtiseven**
Closes #8337 

This PR removed "peak device memory" metric from codebase because there isn't a specific use case for it.

I reviewed all usage of `PEAK_DEVICE_MEMORY`, `peakDevMemory`, `peakDeviceMemory` and `GpuColumnVector.getTotalDeviceMemoryUsed` in our project to ensure that all related code is removed.

### CODE REVIEW

**Comment by thirtiseven** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala [hunk 1]
```diff
@@ -431,8 +410,6 @@ case class GpuOutOfCoreSortIterator(
         while(sortedIt.hasNext) {
           val sortedTbl = sortedIt.next()
           val rows = closeOnExcept(sortedTbl) { _ =>
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala:412]: The closeOnExcept is now overkill, can just be rows = sortedTbl.getRowCuont.toInt.

**Review by jlowe — DISMISSED** [review]
lgtm, minor cleanup suggestion that's not must-fix.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala [hunk 2]
```diff
@@ -661,8 +658,6 @@ class GpuCoalesceIterator(iter: Iterator[ColumnarBatch],
   private def concatBatches(batches: Array[SpillableColumnarBatch]): ColumnarBatch = {
     val wip = batches.safeMap(_.getColumnarBatch())
     val ret = ConcatAndConsumeAll.buildNonEmptyBatchFromTypes(wip, sparkTypes)
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:660]: nit, no need for `ret`

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala [hunk 3]
```diff
@@ -787,7 +779,6 @@ class GpuCompressionAwareCoalesceIterator(
       wip
     }
     val onGPU = ConcatAndConsumeAll.buildNonEmptyBatchFromTypes(toConcat, sparkTypes)
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:781]: nit, no need for `onGPU`

**Review by abellina — APPROVED** [review]
Minor nits

**Comment by thirtiseven** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by abellina — APPROVED** [review]

**Pull request merged**